pkg/rtc/mediatracksubscriptions.go

```diff
@@ -385,7 +385,7 @@ func (t *MediaTrackSubscriptions) RevokeDisallowedSubscribers(allowedSubscriberI
 				"subscriber", subTrack.SubscriberIdentity(),
 				"subscriberID", subTrack.SubscriberID(),
 			)
-			go subTrack.DownTrack().Close()
+			t.RemoveSubscriber(subTrack.SubscriberID(), false)
```

Comment:
This is the main change to ensure that subscriber ID is removed from already subscribed list while revoking.

I did get one failure even after making this change. Added some logging and I could never reproduce in a couple of dozen attempts after that.